### ISSUE TITLE
How to use HTTP server and WebSocket server on the same port ?

### ISSUE THREAD

**Issue by thedub2001** [Links2004/arduinoWebSockets#149]
Hello!
I would like to use the same port for WebSocket server and HTTP server to avoid opening a port on my router.
In the example sketch : [WebSocketServer_LEDcontrol.ino](https://github.com/Links2004/arduinoWebSockets/blob/master/examples/WebSocketServer_LEDcontrol/WebSocketServer_LEDcontrol.ino), is there a way to do something like this ?
``` 
26   ESP8266WebServer server(80); 
27   WebSocketsServer wsServer = WebSocketsServer(80); 
```
When I do that my web browser prints the following error message
``` 
Impossible to establish a connection with the server
``` 
I tried to follow the [#10](https://github.com/Links2004/arduinoWebSockets/issues/10) issue but it's not compiling for the ESP8266

**Comment by quantumlicht** [comment]
I don't think this is possible in your context. The only way it could work is if you run a proxy in front of your application that re-routes to different ports internally.

A quick google search yielded these results. Hopefully this answers your question:
- http://stackoverflow.com/questions/29497725/can-websockets-and-http-server-both-run-on-the-same-port-number
- http://stackoverflow.com/questions/13791050/is-it-possible-to-enable-tcp-http-and-websocket-all-using-the-same-port
- http://serverfault.com/questions/575467/running-a-websocket-server-and-a-http-server-on-the-same-server

**Comment by Links2004** [comment]
the current implementation does not support it direct,
there is a hook when a http connection comes in.
https://github.com/Links2004/arduinoWebSockets/blob/master/src/WebSocketsServer.h#L134

but currently it not be hooked to the ESP8266WebServer.
if you only need to deliver some simple files, overriding the ```handleNonWebsocketConnection``` will do the jop fin.

**Comment by aguaviva** [comment]
I'd love to see any examples showing how to use the web server and the websockets running in the same port

**Comment by kiralikbeyin** [comment]
Any news?

**Comment by dardino** [comment]
no update?
